public/app/plugins/datasource/postgres/response_parser.ts

```diff
@@ -130,9 +134,14 @@ export default class ResponseParser {
     const list = [];
     for (let i = 0; i < table.rows.length; i++) {
       const row = table.rows[i];
+      const timeEnd =
+        timeEndColumnIndex !== -1 && row[timeEndColumnIndex]
+          ? Math.floor(toUtc(row[timeEndColumnIndex]).valueOf())
```

Comment:
Why `toUtc`? Should already be returned as UTC from backend in same manner as `time`.

Reply:
To me it seems like there are some automatic conversions done for column "time", but not "timeend" on backend side. For example, if I add logging to beginning of that method:
`  transformAnnotationResponse(options: any, data: any) {
    console.log (JSON.stringify(data.data.results[options.annotation.name].tables[0]));` 

And then use following Annotation query:
`SELECT
    NOW() - INTERVAL '3 day' AS time,
    NOW() - INTERVAL '1 day' AS timeend,
    'Description' as text,
    'tagA,tagB,tagC' as tags;`

I will get following output:
`{
    "columns": [
        {"text": "time"},
        {"text": "timeend"},
        {"text": "text"},
        {"text": "tags"}
    ],
    "rows": [
        [
            1575709729252.012,
            "2019-12-09T09:08:49.252012Z",
            "Description",
            "tagA,tagB,tagC"
        ]
    ]
}`

You can see how the "time" is 1575709729252.012, but "timeend" is "2019-12-09T09:08:49.252012Z". I followed this result through the frontend code and could see that backend does this conversion somewhere, but I didn't investigate further where in the backend it's done.

Reply:
I'll have a look at the other data sources

Reply:
Added support for MySQL and MSSQL. All 3 behave in similar manner, time is converted on backend and timeend is not.

Reply:
Awesome work :+1: I would really like this logic to reside in the backend because there could be timezone related issues so would suggest applying these changes instead of using the `toUtc` client side:

If you store this in a file you can apply it using `git apply <filename>`. Let me know if you would like me to push the changes instead.

```diff
diff --git a/pkg/tsdb/sqleng/sql_engine.go b/pkg/tsdb/sqleng/sql_engine.go
index 9b3f17f..c481d90 100644
--- a/pkg/tsdb/sqleng/sql_engine.go
+++ b/pkg/tsdb/sqleng/sql_engine.go
@@ -56,6 +56,8 @@ var NewXormEngine = func(driverName string, connectionString string) (*xorm.Engi
        return xorm.NewEngine(driverName, connectionString)
 }
 
+const timeEndColumnName = "timeend"
+
 type sqlQueryEndpoint struct {
        macroEngine            SqlMacroEngine
        queryResultTransformer SqlQueryResultTransformer
@@ -218,6 +220,7 @@ func (e *sqlQueryEndpoint) transformToTable(query *tsdb.Query, rows *core.Rows,
 
        rowCount := 0
        timeIndex := -1
+       timeEndIndex := -1
 
        table := &tsdb.Table{
                Columns: make([]tsdb.TableColumn, columnCount),
@@ -233,6 +236,11 @@ func (e *sqlQueryEndpoint) transformToTable(query *tsdb.Query, rows *core.Rows,
                                break
                        }
                }
+
+               if timeIndex >= 0 && name == timeEndColumnName {
+                       timeEndIndex = i
+                       break
+               }
        }
 
        columnTypes, err := rows.ColumnTypes()
@@ -254,6 +262,7 @@ func (e *sqlQueryEndpoint) transformToTable(query *tsdb.Query, rows *core.Rows,
                // to make native mssql datetime types and epoch dates work in
                // annotation and table queries.
                ConvertSqlTimeColumnToEpochMs(values, timeIndex)
+               ConvertSqlTimeColumnToEpochMs(values, timeEndIndex)
                table.Rows = append(table.Rows, values)
        }
```

Reply:
Thanks for this. I've now added this and removed conversions from frontend